velocity_modelling/cvm/registry.py

```diff
@@ -0,0 +1,827 @@
+        ----------
+        datatype : str
+            The type of data to retrieve (e.g., 'tomography', 'surface', 'basin').
+        name : str
+            The name of the data entry.
+
+        Returns
+        -------
+        dict or None
+            The information dictionary for the specified data entry, or None if not found.
+        """
+        try:
+            data_section = self.registry[datatype]
+        except KeyError:
+            self.log(f"Error: {datatype} not found in registry", VMLogger.ERROR)
+            return None
+
+        for info in data_section:
+            assert "name" in info, (
+                f"Error: This entry in {datatype} has no name defined."
+            )
+            if info["name"] == name:
+                return info
+        self.log(
+            f"Error: {name} for datatype {datatype} not found in registry",
+            VMLogger.ERROR,
+        )
+        raise KeyError(f"{name} not found in {datatype}")
+        return None
+
+    def get_full_path(self, relative_path: Union[Path, str]) -> Path:
+        """
+        Get the full path for a given relative path.
+
+        Parameters
+        ----------
+        relative_path : Path or str
+            The relative path to convert.
+
+        Returns
+        -------
+        Path
+            The full path.
+        """
+        return (
+            DATA_ROOT / relative_path
+            if not Path(relative_path).is_absolute()
+            else Path(relative_path)
+        )
+
+    def load_1d_velo_sub_model(self, v1d_path: Path):
+        """
+        Load a 1D velocity submodel into memory.
+
+        Parameters
+        ----------
+        v1d_path : Path or str
+            The path to the 1D velocity model file.
+
+        Returns
+        -------
+        VelocityModel1D
+            The loaded 1D velocity model data.
+
+        Raises
+        ------
+        SystemExit
+            If the file cannot be found or read.
+        """
+        from velocity_modelling.cvm.velocity1d import (
+            VelocityModel1D,
+        )
+
+        v1d_path = self.get_full_path(v1d_path)
+
+        # Check if the data is already in the cache
+        if v1d_path in self.cache:
+            self.log(f"{v1d_path} loaded from CACHE", VMLogger.DEBUG)
+            return self.cache[v1d_path]
+
+        try:
+            with open(v1d_path, "r") as file:
+                next(file)
+                data = np.loadtxt(file)
+                velo_mod_1d_data = VelocityModel1D(
+                    data[:, 0], data[:, 1], data[:, 2], data[:, 5]
+                )
+                # Store the loaded data in the cache
+                self.cache[v1d_path] = velo_mod_1d_data
+                self.log(f"Loaded 1D velocity model from {v1d_path}", VMLogger.INFO)
+
+        except FileNotFoundError:
+            self.log(
+                f"Error: 1D velocity model file {v1d_path} not found.", VMLogger.ERROR
+            )
+            sys.exit(1)
```

Comment:
Why system exit? why not just raise the error?

Reply:
Fair point. Replaced many sys.exit(1) calls with raise